satpy/writers/cf_writer.py

```diff
@@ -735,10 +735,10 @@ def _try_to_get_crs(new_data):
         if "area" in new_data.attrs:
             if isinstance(new_data.attrs["area"], AreaDefinition):
                 return new_data.attrs["area"].crs
-            # at least one test case passes an area of type str
-            logger.warning(
-                f"Could not tell CRS from area of type {type(new_data.attrs['area']).__name__:s}. "
-                "Assuming projected CRS.")
+            if not isinstance(new_data.attrs["area"], SwathDefinition):
+                logger.warning(
+                    f"Could not tell CRS from area of type {type(new_data.attrs['area']).__name__:s}. "
```

Comment:
I don't want this to hold this PR from being merged, but what are our thoughts on `type(X).__name__` versus `X.__class__.__name__`?

Reply:
What's the difference between the two, except the first is shorter?

Reply:
Nothing. You could consider one more pythonic than the other, but I'm not sure which. Let's keep it this way.